doc/extdev/tutorials/helloworld_ext.rst

```diff
@@ -0,0 +1,150 @@
+.. _exttuto-helloworld:
+
+Developing a "Hello world" directive
+====================================
+
+The objective of this tutorial is to create a very basic extension that adds a new
+directive that outputs a paragraph containing `hello world`.
+
+Only basic information is provided in this tutorial. For more information,
+refer to the :ref:`other tutorials <extensiontutos>` that go into more
+details.
+
+.. warning:: For this extension, you will need some basic understanding of docutils_
+   and Python.
+
+Creating a new extension file
+-----------------------------
+
+Your extension file could be in any folder of your project. In our case,
+let's do the following:
+
+#. Create an :file:`_ext` folder in :file:`source`.
+#. Create a new Python file in the :file:`_ext` folder called
+   :file:`helloworld.py`.
+
+   Here is an example of the folder structure you might obtain:
+
+   .. code-block:: text
+
+         └── source
+             ├── _ext
+             │   └── helloworld.py
+             ├── _static
+             ├── _themes
+             ├── conf.py
+             ├── somefolder
+             ├── somefile.rst
+             └── someotherfile.rst
+
+Writing the extension
+---------------------
+
+Open :file:`helloworld.py` and paste the following code in it:
+
+.. code-block:: python
+
+   from docutils import nodes
+   from docutils.parsers.rst import Directive
+
+
+   class HelloWorld(Directive):
+       def run(self):
+           paragraph_node = nodes.paragraph(text='Hello World!')
+           return [paragraph_node]
+
+
+   def setup(app):
+       app.add_directive("helloworld", HelloWorld)
+
+
+Some essential things are happening in this example, and you will see them
+in all directives:
+
+.. rubric:: Directive declaration
+
+Our new directive is declared in the :code:`HelloWorld` class, it extends
+docutils_' code:`Directive` class. All extensions that create directives
+should extend this class.
+
+.. rubric:: `run` method
+
+This method is a requirement and it is part of every directive. It contains
+the main logic of the directive and it returns a list of docutils nodes to
+be processed by Sphinx.
+
+Read more on this topic in :ref:`exttuto-todo`.
```

Comment:
This would make sense as a `seealso` directive, I think? Also, could you use the `doc` role instead of the `ref` role?

Reply:
I can do that but for my understanding, why do you recommend `doc`? I always go for ref as it allows ppl to move files around with worrying of links.

Reply:
I recommend it because it encourages people to think about the structure of their documentation. It also highlights the fact that there is a cost to moving documentation around without good reason (in the context of broken links etc.)